File: src/main/resources/views/aether/src/entity/vo/GetUserDetailsVo.ts
import UserGroupVo from './UserGroupVo.ts';
import UserPermissionVo from './UserPermissionVo.ts';

/**
 * 用户详情数据
 */
export default interface GetUserDetailsVo {
  /**
   * 用户ID
   */
  id: string;
  
  /**
   * 用户名
   */
  username: string;
  
  /**
   * 用户昵称
   */
  nickname?: string;
  
  /**
   * 用户邮箱
   */
  email?: string;
  
  /**
   * 用户状态
   */
  status?: number;
  
  /**
   * 创建时间
   */
  createTime?: string;
  
  /**
   * 最后登录时间
   */
  lastLoginTime?: string;
  
  /**
   * 用户组列表
   */
  groups?: UserGroupVo[];
  
  /**
   * 用户权限列表
   */
  permissions?: UserPermissionVo[];
} 

File: src/main/java/com/ksptool/ql/biz/model/vo/GetUserDetailsVo.java
package com.ksptool.ql.biz.model.vo;

import lombok.Data;
import java.util.List;

@Data
public class GetUserDetailsVo {
    /**
     * 用户ID
     */
    private Long id;
    
    /**
     * 用户名
     */
    private String username;
    
    /**
     * 用户昵称
     */
    private String nickname;
    
    /**
     * 用户邮箱
     */
    private String email;
    
    /**
     * 用户状态
     */
    private Integer status;
    
    /**
     * 创建时间
     */
    private String createTime;
    
    /**
     * 最后登录时间
     */
    private String lastLoginTime;
    
    /**
     * 用户组列表
     */
    private List<UserGroupVo> groups;
    
    /**
     * 用户权限列表
     */
    private List<UserPermissionVo> permissions;
}
